### PR TITLE
Fix symmetry of mouse movement behaviour

### PR DESCRIPTION
**Pull request by kareltucek**
Solves symmetry problem mentioned in #256 .

I am afraid that this adds a bit of complexity into the code, unfortunately I don't see any way around.

Basically, I am extracting "sign" related switches from 115 and 136 into their own functions, and adding them into the kinetic state.

Please test that up&down scrolling works as expected on Mac.

### CODE REVIEW

File: right/src/usb_report_updater.c [hunk 1]
```diff
@@ -57,6 +61,52 @@ mouse_kinetic_state_t MouseScrollState = {
...
+static void updateDirectionSigns(mouse_kinetic_state_t *kineticState) {
+    updateOneDirectionSign(&kineticState->horizontalStateSign, -1, kineticState->leftState, kineticState->rightState);
+    updateOneDirectionSign(&kineticState->horizontalStateSign,  1, kineticState->rightState, kineticState->leftState);
+    updateOneDirectionSign(&kineticState->verticalStateSign, -1, kineticState->upState, kineticState->downState);
+    updateOneDirectionSign(&kineticState->verticalStateSign,  1, kineticState->downState, kineticState->upState);
+}
+
+// Called on keydown of mouse action. Direction signs ensure that the last pressed action always takes precedence, and therefore
+// have to be updated statefully.
+static void activateDirectionSigns(uint8_t state) {
```
Comment (mondalaci) [inline right/src/usb_report_updater.c:81]: This function could be much simpler by not breaking for every case, but gathering 1 and -1 cases.

Comment (kareltucek) [reply right/src/usb_report_updater.c:81]: I don't see it. How?

Comment (mondalaci) [reply right/src/usb_report_updater.c:81]: It cannot be done, indeed. Sorry, my bad!

Comment (kareltucek) [reply right/src/usb_report_updater.c:81]: :-(

**Pull request merged**